mocks/DDrppi_mocks/countpairs_rp_pi_mocks_kernels.c.src

```diff
@@ -1187,7 +1164,7 @@ static inline int countpairs_rp_pi_mocks_fallback_DOUBLE(const int64_t N0, DOUBL
         const DOUBLE ypos = *y0++;
         const DOUBLE zpos = *z0++;
         for(int w = 0; w < pair.num_weights; w++){
-            pair.weights0[w].d = *local_w0.weights[w]++;
+            pair.weights0[w].d = *(weights0->weights[w] + i);
```

Comment:
This does seem more readable than the `local_w0` pattern; I'm honestly not sure why I wrote it that way in the first place.

Reply:
Yeah me neither but I remember you tracking down a very pernicious bug with the weights and coming to this `local_weights` solution. Regardless, once I saw that this version was producing correct results - I figured this was a bit more readable